Firestore/Swift/Tests/Codable/FirestoreEncoderTests.swift

```diff
@@ -182,6 +182,21 @@ class FirestoreEncoderTests: XCTestCase {
     assertRoundTrip(model: model, encoded: ["date": date])
   }
 
+  func testTimestampCanDecodeAsDate() {
+    struct EncodingModel: Codable, Equatable {
+      let date: Timestamp
+    }
+    struct DecodingModel: Codable, Equatable {
+      let date: Date
+    }
+    let date = Date(timeIntervalSinceReferenceDate: 0)
+    let timestamp = Timestamp(date: date)
+    let model = EncodingModel(date: timestamp)
+    let decoded = DecodingModel(date: date)
+    let encoded = assertEncodes(model, encoded: ["date": timestamp])
```

Comment:
Optional:

change `assertEncodes(_ model: X, encoded: [String: Any])` to `assertEncodes(_ model: X, to expected: [String: Any])`

and `assertDecodes` to `assertDecodes(_ model: [String: Any], to expected: X)`

Reply:
I agree with that recommendation- perhaps can do a followup PR

Reply:
Thanks. I'll do that instead.